engine/src/lib.rs

```diff
@@ -178,20 +178,20 @@ mod contract {
         let mut io = Runtime;
         let state = state::get_state(&io).sdk_unwrap();
         let index = internal_get_upgrade_index();
-        io.return_output(&(index + state.upgrade_delay_blocks).to_le_bytes())
+        io.return_output(&(index).to_le_bytes())
```

Comment:
The parentheses around `index` are redundant. 
```suggestion
        io.return_output(&index.to_le_bytes());
```